geomesa-core/src/main/scala/org/locationtech/geomesa/core/data/AccumuloFeatureStore.scala

```diff
@@ -41,36 +42,50 @@ class AccumuloFeatureStore(val dataStore: AccumuloDataStore, val featureName: Na
     val fids = Lists.newArrayList[FeatureId]()
     if (collection.size > 0) {
       writeBounds(collection.getBounds)
-      writeTimeBounds(collection)
 
+      val minMaxVisitorO = index.getDtgFieldName(collection.getSchema).map { dateField => new MinMaxTimeVisitor(dateField) }
       val fw = dataStore.getFeatureWriterAppend(featureName.getLocalPart, Transaction.AUTO_COMMIT)
 
-      val iter = collection.features()
-      while(iter.hasNext) {
-        val feature = iter.next()
-        val newFeature = fw.next()
-
-        try {
-          newFeature.setAttributes(feature.getAttributes)
-          newFeature.getUserData.putAll(feature.getUserData)
-        } catch {
-          case ex: Exception =>
-            throw new DataSourceException(s"Could not create ${featureName.getLocalPart} out of provided feature: ${feature.getID}", ex)
-        }
+      val updateTimeBounds: SimpleFeature => Unit = { feature => minMaxVisitorO.foreach { _.visit(feature) } }
 
-        val useExisting = java.lang.Boolean.TRUE.equals(feature.getUserData.get(Hints.USE_PROVIDED_FID).asInstanceOf[java.lang.Boolean])
-        if (getQueryCapabilities().isUseProvidedFIDSupported && useExisting) {
-          newFeature.getIdentifier.asInstanceOf[FeatureIdImpl].setID(feature.getID)
+      val write: SimpleFeature => FeatureId =
+        if (minMaxVisitorO.isDefined) { feature =>
+          updateTimeBounds(feature)
+          writeFeature(fw, feature)
         }
+        else { feature => writeFeature(fw, feature) }
```

Comment:
(Picky) Our not-documented-at-all style convention is that functions/work go on newlines.  For example,
```
else { feature => 
  writeFeature(fw, feature) 
}
```